src/main.cpp

```diff
@@ -3731,34 +3739,7 @@ bool CheckBlock(const CBlock& block, CValidationState& state, bool fCheckPOW, bo
             }
         }
     } else {
-        LogPrintf("CheckBlock(DASH) : skipping transaction locking checks\n");
-    }
-
-    // DASH : CHECK MASTERNODE PAYMENTS OR SUPERBLOCKS
-
-    CBlockIndex* pindexPrev = chainActive.Tip();
-    if(pindexPrev != NULL)
-    {
-        int nHeight = 0;
-        if(pindexPrev->GetBlockHash() == block.hashPrevBlock)
-        {
-            nHeight = pindexPrev->nHeight+1;
-        } else {
-            // IF WE CAN'T FIND A SPECIFIC MATCH
-            BlockMap::iterator mi = mapBlockIndex.find(block.hashPrevBlock);
-            if (mi != mapBlockIndex.end() && (*mi).second)
-                nHeight = (*mi).second->nHeight+1;
-        }
-
-        if(nHeight != 0){
-            if(!IsBlockPayeeValid(block.vtx[0], nHeight))
-            {
-                mapRejectedBlocks.insert(make_pair(block.GetHash(), GetTime()));
-                return state.DoS(100, error("CheckBlock() : Couldn't find masternode or superblock payments"));
-            }
-        } else {
-            LogPrintf("CheckBlock() : WARNING: Couldn't find previous block, skipping IsBlockPayeeValid(!)\n");
-        }
```

Comment:
They will pass? :) Actually if you think about it - ConnectBlock _is_ the right place for that piece of code because this way you can receive and store block but you can't connect it to tip until you have enough info e.g. about mnpayment or superblock. You will simply have it stored with "invalid" flag and will be _probably_ able to connect it later. This also means IsBlockValueValid and IsBlockPayeeValid could be adjusted to have even stricter validation (and this probably would be a better solution than the one I propose here, but we can tweak that later), which also would be nice imo.